app/services/alias_updater.rb

```diff
@@ -0,0 +1,50 @@
+class AliasUpdater
+  def initialize(args)
+    @new_index_name = args[:new_index_name]
+    @index_alias = args[:index_alias]
+    @client = args[:client]
+    @logger = args[:logger]
+  end
+
+  def run
+    remove_alias_from_old_index
+    assign_alias_to_new_index
+    logger.info "Alias '#{index_alias}' now pointing to '#{new_index_name}'"
+  end
+
+  private
+
+  attr_reader :logger, :client, :index_alias, :new_index_name
+
+  def remove_alias_from_old_index
+    client.indices.update_aliases body: {
+      actions: [
+        {
+          remove: {
+            index: Dataset.index_name,
+            alias: index_alias
+          }
+        }
+      ]
+    }
+  rescue
```

Comment:
Looks like we're rescuing a very specific kind of error here. How about specifying which error we're rescuing?